src/test/java/seedu/address/model/medicalnote/MedicalNoteTest.java

```diff
@@ -0,0 +1,97 @@
+package seedu.address.model.medicalnote;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+class MedicalNoteTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        // when date is null
+        assertThrows(NullPointerException.class, () -> {
+            new MedicalNote(null, "Lydia Yu", "Patient is asymptomatic.");
+        });
+        // when doctorName is null
+        assertThrows(NullPointerException.class, () -> {
+            new MedicalNote("15-04-2020", null, "Patient is asymptomatic.");
+        });
+        // when content is null
+        assertThrows(NullPointerException.class, () -> {
+            new MedicalNote("15-04-2020", "Lydia Yu", null);
+        });
+    }
+
+    @Test
+    public void constructor_invalidDate_throwsIllegalArgumentException() {
+        String invalidDate = "15/04/20";
```

Comment:
Perhaps it is better to define these variables as named constants (private static final) at the class level.

Reply:
Will that make the code harder to read and be a bit unnecessary? I also made it similar to other classes so maybe we would want to be consistent?